packages/core/src/oidc-utils.ts

```diff
@@ -0,0 +1,85 @@
+import * as actions_http_client from '@actions/http-client'
+import {IRequestOptions} from '@actions/http-client/interfaces'
+import {HttpClient} from '@actions/http-client'
+import {BearerCredentialHandler} from '@actions/http-client/auth'
+import {debug, setSecret} from './core'
+
+interface TokenRequest {
+  aud: string | undefined
+}
+
+interface TokenResponse {
+  value: string | undefined
+}
+
+export class OidcClient {
+
+  private static createHttpClient(allowRetry = true, maxRetry = 10) {
+    let requestOptions: IRequestOptions = {
+      allowRetries: allowRetry,
+      maxRetries: maxRetry
+    }
+    
+    return new HttpClient('actions/oidc-client', [
+      new BearerCredentialHandler(OidcClient.getRuntimeToken())],
+      requestOptions)
+  }
+
+  private static getApiVersion(): string {
+    return '2.0'
+  }
+
+  private static getRuntimeToken(){
+    const token = process.env['ACTIONS_RUNTIME_TOKEN']
+    if (!token) {
+      throw new Error('Unable to get ACTIONS_RUNTIME_TOKEN env variable')
+    }
+    return token
+  }
+
+  private static getIDTokenUrl(){
+    let runtimeUrl = process.env['ACTIONS_ID_TOKEN_REQUEST_URL']
+    if (!runtimeUrl) {
+      throw new Error('Unable to get ACTIONS_ID_TOKEN_REQUEST_URL env variable')
+    }
+    return runtimeUrl + '?api-version=' + OidcClient.getApiVersion()
+  }
+
+  private static async postCall(httpclient: actions_http_client.HttpClient, id_token_url: string, data: TokenRequest): Promise<string> {
```

Comment:
Any reason we need to pass the http client in here? Can we just create it inside here?